File: TypeScript/ShortestPalindrome.ts
function shortestPalindrome(s: string): string {
    let palindrome: string[] = s.split('');
    let palindromeAux: string[] = [...palindrome];
    let j= palindrome.length -1;

    while(!isPalindrome(palindrome))
    {
        palindrome.pop();
    }
    const index = palindrome.length;
    const posfix = palindromeAux.length - index;
    const initialSize = palindromeAux.length;
    for (let i = index; i <  palindromeAux.length; i=i+2)
    {
        palindromeAux.unshift(palindromeAux[i]);
    }
    
    return palindromeAux.join('');
}

function isPalindrome(list: string[]): boolean {
    let j: number = list.length - 1;
    for (let i = 0; i < Math.floor(list.length / 2); i++) {
        if (list[i] !== list[j])
            return false;
        j--;
    }
    return true;
}

File: Python/ShortestPalindrome.py
from typing import List

class Solution:
    def shortestPalindrome(self, s: str) -> str:
        palindrome = list(s)
        palindromeAux = palindrome.copy()
        j= len(palindrome) -1
        while(not self.IsPalindrome(palindrome)):
            palindrome.pop(len(palindrome) -1)
        
        index = len(palindrome)
        i=index
        sizePalindrome = len(palindromeAux)
        while(i < sizePalindrome ):
            palindromeAux.insert(0, palindromeAux[i])
            i = i + 2
            sizePalindrome = len(palindromeAux)
        print(''.join(palindromeAux))
        return ''.join(palindromeAux)
        
    def IsPalindrome(self,lista) -> bool:
        j = len(lista) - 1
        for i in range(len(lista)//2):
            if (lista[i] != lista[j]):
                return False
            j = j-1
        return True